plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResource.java

```diff
@@ -403,20 +412,36 @@ private Answer executeRequest(TrashNetworkVspCommand cmd) {
     private Answer executeRequest(ApplyStaticNatVspCommand cmd) {
         try {
             isNuageVspElementLoaded();
-            _nuageVspElementClient.applyStaticNats(cmd.getNetworkDomainUuid(), cmd.getVpcOrSubnetUuid(), cmd.isL3Network(), cmd.getStaticNatDetails());
+            _nuageVspElementClient.applyStaticNats(cmd.getNetworkDomainUuid(), cmd.getNetworkUuid(), cmd.getVpcOrSubnetUuid(), cmd.isL3Network(),
+                    cmd.isVpc(), cmd.getStaticNatDetails());
             return new ApplyStaticNatVspAnswer(cmd, true, "Applied Static NAT to VSP network mapping to " + cmd.getVpcOrSubnetUuid());
         } catch (Exception e) {
+            s_logger.error("Failure during " + cmd, e);
             return new ApplyStaticNatVspAnswer(cmd, e);
         }
     }
 
+    private Answer executeRequest(ImplementVspCommand cmd) {
+        try {
+            isNuageVspElementLoaded();
+            boolean success = _nuageVspElementClient.implement(cmd.getNetworkId(), cmd.getNetworkDomainUuid(), cmd.getNetworkUuid(), cmd.getNetworkName(), cmd.getVpcOrSubnetUuid(), cmd.isL2Network(),
+                    cmd.isL3Network(), cmd.isVpc(), cmd.isShared(), cmd.getDomainTemplateName(), cmd.isFirewallServiceSupported(), cmd.getDnsServers(), cmd.getIngressFirewallRules(),
+                    cmd.getEgressFirewallRules(), cmd.getAcsFipUuid(), cmd.isEgressDefaultPolicy());
+            return new ImplementVspAnswer(cmd, success, "Implemented network in VSP " + cmd.getNetworkUuid());
+        } catch (Exception e) {
```

Comment:
Our client is using internal exceptions, which are not available in ACS, and which translated to <code>Exception</code> before we return from the client. Therefore we need to catch on <code>Exception</code>